docker_registry.py

```diff
@@ -0,0 +1,212 @@
+                try:
+                    errors = resp.json()['errors']
+                    raise self.DockerRegistryError(errors)
+                except ValueError:
+                    pass
+
+            return resp
+
+    def uploadManifest(self, content, reference=None):
+        """Upload a manifest. Data is given as bytes in content, the digest/tag in reference.
+        If reference is None, the digest is computed and used as reference.
+        On success, the used reference is returned. False otherwise."""
+        content_json = json.loads(content.decode('utf-8'))
+        if "mediaType" not in content_json:
+            raise Exception("Invalid manifest")
+
+        if reference is None:
+            alg = hashlib.sha256()
+            alg.update(content)
+            reference = "sha256:" + alg.hexdigest()
+
+        resp = self.doHttpCall("PUT", "/v2/%s/manifests/%s" % (self.repository, reference),
+                               headers={'Content-Type': content_json["mediaType"]},
+                               data=content)
+
+        if resp.status_code != 201:
+            return False
+
+        return reference
+
+    def uploadManifestFile(self, filename, reference=None):
+        """Upload a manifest. If the filename doesn't equal the digest, it's computed.
+        If reference is None, the digest is used. You can use the manifest's tag
+        for example.
+        On success, the used reference is returned. False otherwise."""
+        with open(filename, "rb") as manifest:
+            content = manifest.read()
+
+            if reference is None:
+                basename = os.path.basename(filename)
+                if basename.startswith("sha256:"):
+                    reference = basename
+
+            if reference is None:
+                raise Exception("No reference determined")
+
+            return self.uploadManifest(content, reference)
+
+
+    def getManifest(self, reference):
+        """Get a (json-parsed) manifest with the given reference (digest or tag).
+        If the manifest does not exist, return None. For other errors, False."""
+        resp = self.doHttpCall("GET", "/v2/%s/manifests/%s" % (self.repository, reference),
+                               headers={'Accept': "application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json"})
+
+        if resp.status_code == 404:
+            return None
+
+        if resp.status_code != 200:
+            return False
+
+        return resp.json()
+
+    def getManifestDigest(self, reference):
+        """Return the digest of the manifest with the given reference.
+        If the manifest doesn't exist or the request fails, it returns False."""
+        resp = self.doHttpCall("HEAD", "/v2/%s/manifests/%s" % (self.repository, reference),
+                               headers={'Accept': "application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json"})
```

Comment:
nit: same thing on quotes, but the same line even :(

Reply:
I always use `'` for keys and `"` for values.

Reply:
my eyes!

Reply:
I suggest to use the style checks provided by flake8 (drop-in replacement for pep8) and follow these :)